pong-server/game-server/pong/consumers.py

```diff
@@ -138,32 +140,29 @@ async def handle_game_message(self, text_data):
     async def pong_message(self, data):
         key = data.get('key')
         is_pressed = data.get('is_pressed', False)
-        player_name = data.get('player_name')
+        sent_player_name = data.get('player_name')
+
+        if key in ['ArrowUp', 'w']:
+            self.up_pressed = is_pressed
+        elif key in ['s', 'ArrowDown']:
+            self.down_pressed = is_pressed
+        speed = -7 * self.up_pressed + 7 * self.down_pressed
+
+        # キーを離したときにすでに逆向きのキーが押されているならspeedの計算はしない
+        def is_change_speed_needed(current_speed):
+            if not is_pressed:
+                if (current_speed == -7 and key in ['ArrowDown', 's']) or (
+                        current_speed == 7 and key in ['ArrowUp', 'w']):
+                    return False
+            return True
 
```

Comment:
```suggestion
        # キーを離したときにすでに逆向きのキーが押されているならspeedの計算はしない
        def is_change_speed_needed(current_speed):
            if not is_pressed:
                if (current_speed == -7 and key in ['ArrowDown', 's']) or (
                        current_speed == 7 and key in ['ArrowUp', 'w']):
                    return False
            return True

```
この部分、成立しないので不要のような気がします（同時押しだとspeed = 0）